crates/sui-types/src/base_types.rs

```diff
@@ -474,8 +474,8 @@ pub fn context_from_digest(digest: TransactionDigest) -> Context {
     // TODO: don't create a HashMap, that wastes memory and costs an allocation!
     let mut carrier = HashMap::new();
     // TODO: figure out exactly what key to use.  I suspect it has to be the parent span ID in OpenTelemetry format.
-    carrier.insert("traceparent".to_string(), hex::encode(digest.0));
-    // carrier.insert("tx_digest".to_string(), hex::encode(digest.0));
+    carrier.insert("traceparent".to_string(), Hex::encode(digest.0));
```

Comment:
These do not have the same behavior as previously. 

Honestly, I got very confused when I realized that somehow in fastcrypto we switched the behavior of 

- previously `hex:encode` didn't attach `0x`, while fastcrypto's `Hex::encode` attaches `0x` 
- previously `encode_bytes_hex` attached `0x`, while fastcrypto's `encode_bytes_hex` doesn't.

IMO the previous behavior was better. It's probably a mistake that our fastcrypto's `Hex::encode` attaches `0x` (this is important for compatibility with other Hex libs + note that the 0x-format is optional for a generic app that uses fastcrypto, and it just happens to be a helper for us).

TL;DR I'd expect the helper function `encode_bytes_hex` to attach the `0x`. We also need better name, cause it's very difficult to spot the difference between the two unctions while reviewing a crate that uses fastcrypto.

My suggestion: fix fastcrypto first, then update Sui, wdyt? @joyqvq @jonas-lj ?

Reply:
https://github.com/MystenLabs/fastcrypto/blob/main/fastcrypto/src/encoding.rs#L155

to be clear `encode_bytes_hex` had always not added 0x and its just `hex::encode`. `Hex::encode` does add 0x. 

there are places we do want 0x and places we dont in Sui, in this PR we follow:

1. when we dont want 0x -> encode_bytes_hex
2. when we want 0x -> Hex::encode

Reply:
For consistency w other hex libs out there and because the 0x is just a fronting convention I'd prefer the struct's main methods to not deal w formats, but only the helper functions should do